tests/cli/test_merge.py

```
@@ -0,0 +1,266 @@
+                list(two_frames),
+                output=str(tmp_path / "merged.gwf"),
+                metadata=[str(_metadata_for(first, tmp_path / "a.yaml"))],
+            )
+
+    def test_a_frame_absent_from_its_metadata_is_refused(self, tmp_path: Path, two_frames) -> None:
+        first, second = two_frames
+        metadata = [
+            str(_metadata_for(first, tmp_path / "a.yaml", digest="")),
+            str(_metadata_for(second, tmp_path / "b.yaml")),
+        ]
+        with pytest.raises(ValueError, match="No hash found in metadata for file a.gwf"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)